modelconverter/__main__.py

```diff
@@ -535,8 +535,17 @@ def archive(
     model_path = resolve_path(path, MODELS_DIR)
     cfg = archive_from_model(model_path)
     save_path = save_path or f"{cfg.model.metadata.name}.tar.xz"
-    if not save_path.endswith(".tar.xz"):
-        save_path += f"/{cfg.model.metadata.name}.tar.xz"
+    if save_path.endswith("tar.xz"):
+        compression = "xz"
+    elif save_path.endswith("tar.gz"):
+        compression = "gz"
+    elif save_path.endswith("tar.bz2"):
+        compression = "bz2"
+    else:
+        compression = "xz"
+
+    if not save_path.endswith(f".tar.{compression}"):
+        save_path += f"/{cfg.model.metadata.name}.tar.{compression}"
```

Comment:
Could it happen that it's already named `{cfg.model.metadata.name}` but is just missing `.tar.{compression}`?

Reply:
That's possible, though I'm not sure what would be the right approach here. Assume the user meant to write `archive_name.tar.xz` or they actually did want to save it to `archice_name/archive_name.tar.xz`? I'd vouch for the second, I can imagine you could want to have a directory `resnet18` and inside be saving `resnet18.dlc`, `resnet18.blob` and `resnet18.tar.xz`.

Reply:
That makes sense, yes.